Moved Data location from Buckets->Reference. Added documentation for upcoming Jurisdictions (#10580)
cloudflare/cloudflare-docs: content/r2/reference/data-location.md

## Jurisdictional Restrictions

{{<Aside type="note">}}

This feature is currently in beta. If you have feedback, reach out to us on the [Cloudflare Developer Discord](https://discord.gg/rrZXVVcKQF) in the #r2-storage channel or open a thread on the [Community Forum](https://community.cloudflare.com/c/developers/storage/81).

{{</Aside>}}

Jurisdictional Restrictions guarantee objects in a bucket are stored within a specific jurisdiction.

Use Jurisdictional Restrictions when you need to ensure data is stored and processed within a jurisdiction to meet data residency requirements, including local regulations such as the [GDPR](https://gdpr-info.eu/).

### Set jurisdiction via the Cloudflare dashboard

1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/) and select R2.
2. Select **Create bucket**.
3. Enter a name for the bucket.
4. Under **Location**, select **Specify jurisdiction** and choose a jurisdiction from the list.
5. Select **Create bucket** to complete the bucket creation process.

### Using jurisdictions from Workers

> To access R2 buckets that belong to a jurisdiction from [Workers](/workers/), you'll need to specify the jurisdiction as well as the bucket name as part of your [bindings](/r2/api/workers/workers-api-usage/#3-bind-your-bucket-to-a-worker) in your `wrangler.toml`:

Reviewer: ```suggestion
To access R2 buckets that belong to a jurisdiction from [Workers](/workers/), you will need to specify the jurisdiction as well as the bucket name as part of your [bindings](/r2/api/workers/workers-api-usage/#3-bind-your-bucket-to-a-worker) in your `wrangler.toml`:
```

```toml
[[r2_buckets]]
bindings = [
  { binding = "MY_BUCKET", bucket_name = "<YOUR_BUCKET_NAME>", jurisdiction = "<JURISDICTION>" }
]
```

> For more information on getting started visit to [Use R2 from Workers](/r2/api/workers/workers-api-usage/). 

Reviewer: ```suggestion
For more information on getting started, refer to [Use R2 from Workers](/r2/api/workers/workers-api-usage/). 
```

The example below shows how to create an R2 bucket in the `eu` jurisdiction using the [`@aws-sdk/client-s3`](https://www.npmjs.com/package/@aws-sdk/client-s3) package for JavaScript.
```js
import { S3Client, CreateBucketCommand } from "@aws-sdk/client-s3";
const S3 = new S3Client({
	endpoint: "https://4893d737c0b9e484dfc37ec392b5fa8a.eu.r2.cloudflarestorage.com",
	credentials: {
		accessKeyId: "7dc27c125a22ad808cd01df8ec309d41",
		secretAccessKey:
			"1aa5c5b0c43defdb88f567487c071d17e234126133444770a706ae09336c57a4",
	},
	region: "auto",
});
await S3.send(
  new CreateBucketCommand({
    Bucket: "YOUR_BUCKET_NAME"
  })
);
```

Refer to [Examples](/r2/examples/) for additional examples from other S3 SDKs.

### Available jurisdictions

The following jurisdictions are supported:

| Jurisdiction | Jurisdiction description      |
| ---- | --------------------- |
| eu | European Union |

### Limitations

During the beta, the following services will not interact with R2 resources with assigned jurisdictions:
* [Super Slurper](/r2/data-migration/)
* [Logpush](/logs/get-started/enable-destinations/r2/)

### Additional considerations

> Once an R2 bucket is created, the jurisdiction can't be changed.

Reviewer: ```suggestion
Once an R2 bucket is created, the jurisdiction cannot be changed.
```
